### PR TITLE
Update substrate to 2.0.1

### PR DESCRIPTION
**Pull request by dvdplm**
Brings a bunch of other updates, `kvdb`, `kvdb-rocksdb`, `parity-util-mem` and friends.

Biggest point of uncertainty is the errors, `BlockchainError` lost the `Msg` catch-all member so it takes some creativity to match up what we have here to what's available. :/

### CODE REVIEW

**Review by emostov — APPROVED** [review]
lgtm

> Biggest point of uncertainty is the errors, BlockchainError lost the Msg catch-all member so it takes some creativity to match up what we have here to what's available. :/

Updated errors look reasonable to me. Should we note the error update as a breaking change so we can bump relevant versioning on next release?

Also whats up with the CI, does it need fmt again?

**Comment by niklasad1** [comment]
> Also whats up with the CI, does it need fmt again?

No, it looks like we are using two different versions of `parity-util-mem`. `sc-service` requires 0.7.x and `substrate-archive-backend` is using `0.8`.

I think it goes away once you downgrade the substrate dependencies in `substrate-archive-backend` to `2.0.1` @dvdplm :)

**Comment by dvdplm** [comment]
> I think it goes away once you downgrade the substrate dependencies in `substrate-archive-backend`, `substrate-archive-common` and `substrate-archive` to `2.0.1` @dvdplm :)

You surely mean upgrade? The only way is up!

(Yes, you are totally correct; I missed that the binaries are not in the workspace, ty.)

**Review by insipx — APPROVED** [review]

File: substrate-archive/src/database/listener.rs [hunk 1]
```diff
@@ -149,9 +150,9 @@ where
 			}
 			// collect the rest of the results, before exiting, as long as the collection completes
 			// in a reasonable amount of time
-			let timeout = smol::Timer::after(std::time::Duration::from_secs(1));
+			let timeout = smol::Timer::after(Duration::from_secs(1));
 			futures::select! {
-				_ = timeout.fuse() => {},
+				_ = FutureExt::fuse(timeout) => {},
```
Comment (niklasad1) [inline substrate-archive/src/database/listener.rs:155]: What's the reason behind this change?

Comment (insipx) [reply substrate-archive/src/database/listener.rs:155]: iirc won't compile because it complains that .fuse() might point to the `StreamExt`, `FutureExt` or other `fuse()` method

Comment (dvdplm) [reply substrate-archive/src/database/listener.rs:155]: Indeed. But the deeper reason the compiler can't see this escapes me.
